hs_core/models.py

```diff
@@ -1791,6 +1792,16 @@ def get_xml(self, pretty_print=True):
         rdfs1_isDefinedBy = etree.SubElement(rdf_Description_resource, '{%s}isDefinedBy' % self.NAMESPACES['rdfs1'])
         rdfs1_isDefinedBy.text = current_site_url() + "/terms"
 
+        # encode extended key/value arbitrary metadata
+        resource = BaseResource.objects.filter(object_id=self.id).first()
+        for key in resource.extra_metadata:
+            hsterms_key_value = etree.SubElement(rdf_Description, '{%s}extendedMetadata' % self.NAMESPACES['hsterms'])
+            hsterms_key_value_rdf_Description = etree.SubElement(hsterms_key_value, '{%s}Description' % self.NAMESPACES['rdf'])
+            hsterms_key = etree.SubElement(hsterms_key_value_rdf_Description, '{%s}key' % self.NAMESPACES['hsterms'])
+            hsterms_key.text = key
+            hsterms_value = etree.SubElement(hsterms_key_value_rdf_Description, '{%s}value' % self.NAMESPACES['hsterms'])
+            hsterms_value.text = resource.extra_metadata[key]
+
```

Comment:
@alvacouch @pkdash I believe extra_metadata being added here does not have anything to do with AbstractMetadataElement, but instead just a HStoreField defined on our resource model. The code above just serialize this extra_metadata dict field into xml. This extra HStore dictionary field allows us to attach an arbitrary key-value dict to each resource for defining extra unconstrained metadata for resources.